cpp/tests/utilities/column_wrapper.cuh

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright (c) 2018, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef COLUMN_WRAPPER_H
+#define COLUMN_WRAPPER_H
+
+#include <thrust/equal.h>
+#include <thrust/logical.h>
+#include <bitset>
+#include "cudf.h"
+#include "rmm/rmm.h"
+#include "utilities/bit_util.cuh"
+#include "utilities/type_dispatcher.hpp"
+
+#ifndef CUDA_RT_CALL
+#define CUDA_RT_CALL(call)                                                    \
+  do {                                                                        \
+    cudaError_t cudaStatus = (call);                                          \
+    if (cudaSuccess != cudaStatus) {                                          \
+      fprintf(stderr,                                                         \
+              "ERROR: CUDA RT call \"%s\" in line %d of file %s failed with " \
+              "%s (%d).\n",                                                   \
+              #call, __LINE__, __FILE__, cudaGetErrorString(cudaStatus),      \
+              cudaStatus);                                                    \
+      exit(1);                                                                \
+    }                                                                         \
+  } while (0)
+#endif
+
+namespace cudf {
+namespace test {
+
+template <typename ColumnType>
+struct column_wrapper {
```

Comment:
This file and this struct both need block comments explaining the functionality.

Reply:
Also why is this a struct and not a class?

Reply:
I notice that this struct is not yet used anywhere. I don't feel completely comfortable reviewing it without any examples of usage (especially without docs). What is the intended use?

Reply:
It's purpose is to provide a more convenient abstraction for a `gdf_column` specifically for unit testing by adding constructors that accept lambda initializers, vector intializers, copying the column's data from device to host, check for equality between two columns, etc. 

It's used in my scatter/gather unit tests. I added this in a separate PR to avoid bloat in https://github.com/rapidsai/cudf/pull/577

https://github.com/rapidsai/cudf/pull/577/files#diff-4b776ff337347f647a6b0d1a91092d2b

Reply:
I'm aware of the difference between struct and class. :)